File: Java/src/_1833_Maximum_Ice_Cream_Bars/Solution.java
package _1833_Maximum_Ice_Cream_Bars;

import java.util.Arrays;

/**
 * Date: 2021/7/2 9:44
 */
public class Solution {

    public int maxIceCream(int[] costs, int coins) {
        int n = costs.length;
        Arrays.sort(costs);
        int cnt = 0;
        while (cnt < n && costs[cnt] <= coins) coins -= costs[cnt++];
        return cnt;
    }

    public static void main(String[] args) {

        System.out.println((new Solution()).maxIceCream(new int[]{1, 3, 2, 4, 1}, 7));  // 4
    }
}

File: Python/_1833_Maximum_Ice_Cream_Bars/Solution.py
# coding=utf-8
# Date: 2021/7/2 9:53
from typing import List


# 执行用时：196 ms, 在所有 Python3 提交中击败了18.00%的用户
# 内存消耗：24.7 MB, 在所有 Python3 提交中击败了84.29%的用户
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        n = len(costs)
        costs.sort()
        cnt = 0
        while cnt < n and costs[cnt] <= coins:
            coins -= costs[cnt]
            cnt += 1
        return cnt


if __name__ == "__main__":
    print(Solution().maxIceCream([1, 3, 2, 4, 1], 7))
